test/libsolidity/GasCosts.cpp

```diff
@@ -113,9 +113,8 @@ BOOST_AUTO_TEST_CASE(string_storage)
 				CHECK_DEPLOY_GAS(0, 109237, evmVersion);
 			else if (evmVersion < EVMVersion::shanghai())
 				CHECK_DEPLOY_GAS(0, 97693, evmVersion);
-			// Shanghai is cheaper due to `push0`
 			else
-				CHECK_DEPLOY_GAS(0, 97067, evmVersion);
+				CHECK_DEPLOY_GAS(0, 97715, evmVersion);
```

Comment:
The increase in gas is due to the reduced cost of `PUSH0`. Before, there was a `PUSH <tag> JUMP` which was not inlined because of its cost, but with the introduction of `PUSH0` it is now cheaper enough to be inlined. Previously, the "pushtag-jump" pattern was removed because it was next to the very tag it was jumping to. But now with the code inlined, that removal does not happen.